app/javascript/tests/subject/CaseStatus.test.js

```diff
@@ -0,0 +1,225 @@
+        expect(wrapper.find(Modal.Title).exists()).toBeTruthy();
+        expect(wrapper.find(Modal.Title).text()).toEqual('Case Status');
+        expect(wrapper.find(Modal.Body).exists()).toBeTruthy();
+        expect(wrapper.find(Modal.Footer).exists()).toBeTruthy();
+        expect(wrapper.find(Button).at(0).text()).toEqual('Cancel');
+        expect(wrapper.find(Button).at(1).text()).toEqual('Submit');
+    });
+
+    it('Properly renders modal body and updates state when not monitoring', () => {
+        const wrapper = getWrapper(mockPatient3, false);
+        wrapper.find(Form.Control).simulate('change', { target: { id: 'case_status', value: 'Confirmed' }, persist: jest.fn() });
+        const modalBody = wrapper.find(Modal.Body);
+
+        expect(wrapper.state('showCaseStatusModal')).toBeTruthy();
+        expect(wrapper.state('case_status')).toEqual('Confirmed');
+        expect(wrapper.state('confirmedOrProbable')).toBeTruthy();
+        expect(modalBody.find('p').text()).toEqual(`Are you sure you want to change case status from ${mockPatient3.case_status} to Confirmed? Since this record is on the Closed line list, updating this value will not move this record to another line list. If this individual should be actively monitored, please update the record’s Monitoring Status.`);
+    });
+
+    it('Properly renders modal body and updates state when changing Case Status to blank', () => {
+        const wrapper = getWrapper(mockPatient1, false);
+        wrapper.find(Form.Control).simulate('change', { target: { id: 'case_status', value: '' }, persist: jest.fn() });
+        const modalBody = wrapper.find(Modal.Body);
+
+        expect(wrapper.state('showCaseStatusModal')).toBeTruthy();
+        expect(wrapper.state('case_status')).toEqual('');
+        expect(wrapper.state('confirmedOrProbable')).toBeFalsy();
+        expect(modalBody.find('p').text()).toEqual(`Are you sure you want to change case status from ${mockPatient1.case_status} to blank? The monitoree will remain in the same workflow.`);
+    });
+
+    it('Properly renders modal body and updates state when changing Case Status to Suspect, Unknown or Not a Case from Confirmed or Probable in isolation workflow', () => {
+        const wrapper = getWrapper(mockPatient4, false);
+        wrapper.find(Form.Control).simulate('change', { target: { id: 'case_status', value: 'Unknown' }, persist: jest.fn() });
+        const modalBody = wrapper.find(Modal.Body);
+
+        expect(wrapper.state('showCaseStatusModal')).toBeTruthy();
+        expect(wrapper.state('case_status')).toEqual('Unknown');
+        expect(wrapper.state('confirmedOrProbable')).toBeFalsy();
+        expect(modalBody.find('p').text()).toEqual('This case will be moved to the exposure workflow and will be placed in the symptomatic, non-reporting, or asymptomatic line list as appropriate to continue exposure monitoring.');
+    });
```

Comment:
This test _should_ be failing currently (see related comment above), but it's not because it doesn't test the `isolation` state variable which we should be verifying is set to false here.

Reply:
done